src/implementations/token-registry/role/set-role.test.ts

```diff
@@ -0,0 +1,101 @@
+import { TradeTrustERC721__factory } from "@govtechsg/token-registry/contracts";
+import { Wallet } from "ethers";
+import { TokenRegistrySetRoleCommand } from "../../../commands/token-registry/token-registry-command.type";
+import { setRoleToTokenRegistry } from "./set-role";
+import { connectToTokenRegistry } from "../helpers";
+
+jest.mock("@govtechsg/token-registry/contracts");
+jest.mock("../helpers", () => {
+  const originalModule = jest.requireActual("../helpers");
+  return {
+    __esModule: true,
+    ...originalModule,
+    connectToTokenRegistry: jest.fn(),
+  };
+});
+
+const roleParams: TokenRegistrySetRoleCommand = {
+  address: "0x1122",
+  role: "0x1",
+  adminRole: "0x2",
+  network: "ropsten",
+  gasPriceScale: 1,
+  dryRun: false,
+};
+
+// TODO the following test is very fragile and might break on every interface change of TradeTrustERC721Factory
+// ideally must setup ganache, and run the function over it
+describe("token-registry", () => {
+  describe("set role for token registry", () => {
+    const mockedTradeTrustERC721Factory: jest.Mock<TradeTrustERC721__factory> = TradeTrustERC721__factory as any;
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore mock static method
+    const mockedConnectERC721: jest.Mock = connectToTokenRegistry as jest.Mock;
+    const mockGrantRole = jest.fn();
+    const mockCallStaticGrantRole = jest.fn().mockResolvedValue(undefined);
+
+    beforeEach(() => {
+      delete process.env.OA_PRIVATE_KEY;
+      mockedTradeTrustERC721Factory.mockReset();
+      mockedConnectERC721.mockReset();
+
+      mockGrantRole.mockReturnValue({
+        hash: "hash",
+        wait: () => Promise.resolve({ transactionHash: "transactionHash" }),
+      });
+
+      mockedConnectERC721.mockReturnValue({
+        setRoleAdmin: mockGrantRole,
+        callStatic: {
+          setRoleAdmin: mockCallStaticGrantRole,
+        },
+      });
+      mockGrantRole.mockClear();
+      mockCallStaticGrantRole.mockClear();
+    });
+
+    it("should take in the key from environment variable", async () => {
+      process.env.OA_PRIVATE_KEY = "0000000000000000000000000000000000000000000000000000000000000002";
+
+      await setRoleToTokenRegistry(roleParams);
+
+      const passedSigner: Wallet = mockedConnectERC721.mock.calls[0][0]["wallet"];
+      expect(passedSigner.privateKey).toBe(`0x${process.env.OA_PRIVATE_KEY}`);
+    });
```

Comment:
i know this is copy pastaed from the other tests, but can we not test this here? it bloats this test file what do you think?